src/storage.rs

```diff
@@ -36,16 +70,99 @@ impl Storage {
     /// # Description
     /// Closes all file descriptors
     /// # Examples
+
+    // @TODO specify more useful error type
     pub fn close(&mut self) -> Result<(), ()> {
         // @TODO implement
         Ok(())
     }
+
+    /// # Description
+    /// Blobs count contains closed blobs and one opened, if is some
+    /// # Examples
+    /// ```
+    /// use pearl::Builder;
+    ///
+    /// let mut storage = Builder::new().work_dir("/tmp/pearl/").build();
+    /// storage.init();
+    /// assert_eq!(storage.blobs_count(), 1);
+    /// ```
+    pub fn blobs_count(&self) -> usize {
+        self.blobs.len() + if self.opened_blob.is_some() { 1 } else { 0 }
+    }
+}
+
+impl Storage {
+    fn prepare_work_dir(&self) -> io::Result<()> {
+        let wd = Path::new(&self.config.work_dir);
+        if let Err(e) = fs::read_dir(wd) {
+            match e.kind() {
+                ErrorKind::NotFound => {
+                    error!("\"{}\" not found", self.config.work_dir);
+                }
+                _ => return Err(e),
+            }
+        }
+        debug!(
+            "create work dir recursively: {}",
+            wd.to_str().unwrap_or("failed to convert path to str")
+        );
+        fs::create_dir_all(wd)?;
+        Ok(())
+    }
+
+    // @TODO specify more useful error type
+    #[inline]
+    fn init_opened_blob(&mut self) -> Result<(), ()> {
+        self.opened_blob = Box::new(Some(Default::default()));
+        Ok(())
+    }
+
+    // @TODO specify more useful error type
+    fn init_new(&mut self) -> Result<(), ()> {
+        self.init_opened_blob()
+    }
+
+    // @TODO specify more useful error type
+    fn init_from_existing(&mut self, files: Vec<DirEntry>) -> Result<(), ()> {
+        self.blobs = files
```

Comment:
enumerating files from disk could have unpredictable order. we need to extract some id we will embed into file name example: pblob-1, pblob-2 and then sort it by this name in code.

Reply:
I enumerating `self.blobs: Vec<Blobs>` and indexing the same `self.blobs`. Indexing blobs by id from their name marked as TODO